src/transformers/cache_utils.py

```diff
@@ -184,6 +187,120 @@ def from_legacy_cache(cls, past_key_values: Optional[Tuple[Tuple[torch.FloatTens
         return cache
 
 
+class QuantCache(Cache):
+    def __init__(self, n_bits: int = 2, q_group_size: int = 64, residual_length: int = 128) -> None:
+        self.key_cache: List[torch.Tensor] = []
+        self.value_cache: List[torch.Tensor] = []
+        self._key_cache_quant: List[torch.Tensor] = []
+        self._value_cache_quant: List[torch.Tensor] = []
+        self.seen_token = 0
+        self.residual_length = residual_length
+        self.qtype = qint4 if n_bits == 4 else qint2
```

Comment:
we should check whether `n_bits` is a valid input -- only `2` and `4` seem to be valid inputs here

Reply:
Ah yes, we validate in generation config but now I see that "validate()" is not called when updating. Will add check in Cache also :)